packages/dds/tree/src/shared-tree/sharedTree.ts

```diff
@@ -123,28 +122,17 @@ export interface ISharedTreeCheckout extends AnchorLocator {
 		 * Close this transaction by squashing its edits and committing them as a single edit. If this is the root local branch and there are no
 		 * ongoing transactions remaining, the squashed edit will be submitted to Fluid.
 		 */
-		commit: () => void;
+		commit: () => TransactionResult.Commit;
```

Comment:
What is the purpose of this change?

Reply:
I assume its for ergononomics, so you can do `return commit();` or `return abort();` in some cases. That said the cases that could leverage this are not, I'll update them

Also I don't like making these all writable members instead of readonly function or methods! I'll fix that too.

Reply:
Done.